lib/elixir/lib/access.ex

```diff
@@ -329,43 +307,13 @@ defmodule Access do
   """
   @spec get(container, term, term) :: term
   @spec get(nil_container, any, default) :: default when default: var
-  def get(container, key, default \\ nil)
-
-  def get(%module{} = container, key, default) do
-    try do
-      module.fetch(container, key)
-    rescue
-      exception in UndefinedFunctionError ->
-        raise_undefined_behaviour(exception, module, {^module, :fetch, [^container, ^key], _})
-    else
+  def get(container, key, default \\ nil) do
+    case fetch(container, key) do
```

Comment:
I think the gain in code simplicity is pretty big here against the very slight optimization of not calling `fetch/2` internally (we're losing something like 25 lines of code after all), but I can revert if necessary.

Reply:
I would keep it for lists and maps. It's called very often in most programs, so even a small optimisation matters.
Custom Access implementations, on the other hand, are probably not as common.

Reply:
Exactly what @michalmuskala said. This is called **a lot**, the more verbose code here is justified.